File: BACKEND/TM.Domain/Interface/ITaskItemRepository.cs
using TM.Domain.Entity;

namespace TM.Domain.Interface;

public interface ITaskItemRepository
{
    Task<bool> CreateTaskItem(TaskItem taskItem, CancellationToken cancellationToken);
    Task<bool> UpdateTaskItem(TaskItem taskItem, CancellationToken cancellationToken);
    Task<bool> RemoveTaskItem(int id, CancellationToken cancellationToken);
    Task<TaskItem> GetTaskItemById(int id, CancellationToken cancellationToken);
    Task<List<TaskItem>> GetAllTaskItem(CancellationToken cancellationToken);
}

File: FRONTEND/TM.Frontend/src/domain/interface/ITaskItemRepository.ts
import type { CreateTaskInput } from '../../application/task/createTaskInput';
import type { UpdateTaskInput } from '../../application/task/updateTaskInput';
import type { TaskItem } from '../entity/TaskItem';

export interface ITaskItemRepository {
    createTaskItem(taskItem:CreateTaskInput) : Promise<boolean>;  
    updateTaskItem(taskItem:UpdateTaskInput) : Promise<boolean>;
    removeTaskitem(id:number) : Promise<void>;
    getTaskItemById(id:number) : Promise<TaskItem>;
    getAllTaskItem() : Promise<TaskItem[]>;
}